File: Round 762/B_Squares_and_Cubes.cpp
#include <iostream>
#include <math.h> 
using namespace std;

int countSC(int N)
{
    int res = (int)sqrt(N) + (int)cbrt(N) - (int)(sqrt(cbrt(N)));

    return res;
}

// Driver code
int main()
{
    long long T;
    cin >> T;
    while (T--){
        long long N;
        cin >> N;
        cout <<countSC(N) << endl;
        
    }
}

File: Round 762/B_Squares_and_Cubes.py
import math  # for sqrt()

# Function to count the number
# of perfect squares and cubes


def countSC(N):
    res = (int(math.sqrt(N)) +
           int(N ** (1 / 3)) -
           int(math.sqrt(N ** (1 / 3))))

    return res


# Driver code




# Driver code
for q in range(int(input())):
    n = int(input())

    print(countSC(n+1))
